pkgs/development/python-modules/importlib-metadata/2.nix

```diff
@@ -0,0 +1,44 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools-scm
+, zipp
+, pathlib2
+, contextlib2
+, configparser
+, isPy3k
+, importlib-resources
+, packaging
+}:
+
+buildPythonPackage rec {
+  pname = "importlib-metadata";
+  version = "2.1.1";
+
+  src = fetchPypi {
+    pname = "importlib_metadata";
+    inherit version;
+    sha256 = "1pdmsmwagimn0lsl4x7sg3skcr2fvzqpv2pjd1rh7yrm5gzrxpmq";
+  };
+
+  nativeBuildInputs = [ setuptools-scm ];
+
+  propagatedBuildInputs = [ zipp ]
+    ++ lib.optionals (!isPy3k) [ pathlib2 contextlib2 configparser ];
+
+  doCheck = false; # Cyclic dependencies.
```

Comment:
I thought removing one test gets rid of the cyclic dependency.

Reply:
I tried to remove `tests/test_main.py` which requests `pyflakefs` but there is more. For me it looks like that the circle is `importlib-metadata` -> `pluggy` -> `pytest` -> `importlib-metadata`.

Reply:
If you keep `doCheck = false`, you shouldn't remove any files though.

Reply:
Since importlib-metadata is built-in in python >=3.8, perhaps we could at least run the checks on python>=3.8?  Even though importlib-metadata is probably not that useful as a separate package for python>=3.8, it might still smoke out problems that would also be present for earlier python versions.

It looks like pluggy doesn't have the dependency for python>=3.8 (https://github.com/pytest-dev/pluggy/blob/0.13.1/setup.py#L40), so presumably the pluggy nix expression could be updated to break the cycle in that case.